File: lib/models/card_details.dart
// @author nateshmbhat created on 30,June,2020

import 'package:card_scanner/utils.dart';

class CardDetails {
  String _cardNumber = "";
  String _cardIssuer = "";
  String _cardHolderName = "";
  String _expiryDate = "";

  CardDetails.fromMap(Map<String, String> map) {
    _cardNumber = map['cardNumber'] ?? '';
    _cardIssuer = CardUtils().getCardIssuer(_cardNumber).toString();
    _cardHolderName = map['cardHolderName'] ?? '';
    _expiryDate = map['expiryDate'] ?? '';
  }

  Map<String, String> get map => {
        'cardNumber': _cardNumber,
        'cardIssuer': _cardIssuer,
        'cardHolderName': _cardHolderName,
        'expiryDate': _expiryDate,
      };

  @override
  String toString() {
    var string = '';
    string += _cardNumber.isEmpty ? "" : 'Card Number = $cardNumber\n';
    string += _expiryDate.isEmpty ? "" : 'Expiry Date = $expiryDate\n';
    string += _cardIssuer.isEmpty ? "" : 'Card Issuer = $cardIssuer\n';
    string += _cardHolderName.isEmpty ? "" : 'Card Holder Name = $cardHolderName\n';
    return string;
  }

  String get cardNumber => _cardNumber;

  String get cardIssuer => _cardIssuer;

  String get cardHolderName => _cardHolderName;

  String get expiryDate => _expiryDate;
}

File: ios/Classes/Scanner Core/Models/CardDetails.swift
//
//  CardDetails.swift
//  card_scanner
//
//  Created by Abhinav Kumar on 12/02/21.
//

import Foundation

public class CardDetails: CustomStringConvertible {
    var cardNumber: String = ""
    var cardHolderName: String = ""
    var expiryDate: String = ""
    
    init(cardNumber: String = "", cardHolderName: String = "", expiryDate: String = "") {
        self.cardNumber = cardNumber
        self.cardHolderName = cardHolderName
        self.expiryDate = expiryDate
    }
    
    init(from options: [String: String]?) {
        if let options = options {
            if let cardNumber = options["cardNumber"] {
                self.cardNumber = cardNumber
            }
            
            if let cardHolderName = options["cardHolderName"] {
                self.cardHolderName = cardHolderName
            }
            
            if let expiryDate = options["expiryDate"] {
                self.expiryDate = expiryDate
            }
        }
    }
    
    public var description: String {
        return """
        Card Number: \(cardNumber)
        Card Holder Name: \(cardHolderName)
        Expiry Date: \(expiryDate)
        """
    }
    
    var dictionary: [String: String] {
        return [
            "cardNumber": cardNumber,
            "cardHolderName": cardHolderName,
            "expiryDate": expiryDate,
            "cardIssuer": ""
        ]
    }
}
